test/Microsoft.AspNet.Mvc.Core.Test/Formatters/JsonPatchInputFormatterTest.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.JsonPatch;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class JsonPatchInputFormatterTest
+    {
+        [Fact]
+        public async Task JsonPatchInputFormatter_ReadsOneOperation_Successfully()
+        {
+            // Arrange
+            var formatter = new JsonPatchInputFormatter();
+            var content = "[{\"op\":\"add\",\"path\":\"Customer/Name\",\"value\":\"John\"}]";
+            var contentBytes = Encoding.UTF8.GetBytes(content);
+
+            var actionContext = GetActionContext(contentBytes);
+            var metadata = new EmptyModelMetadataProvider().GetMetadataForType(typeof(JsonPatchDocument<Customer>));
```

Comment:
I don't think you need the MMP for anything here. Just pass in the type to the `InputFormatterContext`